Common/Chairloader/IConfigNodeV1.h

```diff
@@ -0,0 +1,120 @@
+//
+// Created by theli on 4/26/2025.
+//
+
+#ifndef ICONFIGNODEV1_H
+#define ICONFIGNODEV1_H
+#include <memory>
+#include <optional>
+#include <pugixml.hpp>
+#include <stdexcept>
+#include <variant>
+#include "Chairloader/ConfigNodeRef.h"
+
+// interface for version 1 of config node
+struct IConfigNodeV1 {
+    enum class NodeType { String, Bool, Int, Uint, Int64, Uint64, Float, XmlNode, Enum, Unknown };
```

Comment:
I think `NodeType` can be renamed to `EConfigNodeType`, placed in the global namespace and shared between all future versions. In case something needs to be removed, it can be renamed to `DEPRECATED_something` without breaking much.

`Unknown` should be the first value so it's the default (zero) when initializaing.